src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -2011,6 +2011,7 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
             {
                 // Delegate targets are always treated as direct calls here. (It would be nice to clean it up...).
                 directCall = true;
+                resolvedCallVirt = !targetMethod.IsVirtual || targetMethod.IsFinal || targetMethod.OwningType.IsSealed();
```

Comment:
It works in CoreCLR because of it can deal with the virtual call that is not actually one till pretty late and then compensate for it. I do not think it makes sense to do it this way for CoreRT. Also - it make sense to not diverge JIT-EE interface implementation from CoreCLR for no good reason, but I would not write convoluted code to make it match. 

This is actually my fault. I have added this hack for delegates when I was refactoring the JIT-EE interface a few years ago but forgot to remove it (TFS changeset 13508).

I have submitted #1106 to cleanup this up.